data/saas/saas_connector_registry.toml

```diff
@@ -159,3 +159,9 @@ config = "data/saas/config/recharge_config.yml"
 dataset = "data/saas/dataset/recharge_dataset.yml"
 icon = "data/saas/icon/default.svg"
 human_readable = "Recharge"
+
+[universal_analytics]
+config = "data/saas/config/universal_analytics_config.yml"
+dataset = "data/saas/dataset/universal_analytics_dataset.yml"
+icon = "data/saas/icon/google_analytics.svg"
+human_readable = "Universal Analytics"
```

Comment:
The work here adds a new "Universal Analytics" connector instead of adapting the existing Google Analytics connector to support both GA4 and UA.

This seems to be the quickest way forward because we don't seem to have options yet for marking configuration fields as "optional" in saas connectors or to add a lot of conditional logic to the connectors.   The saas connectors largely run the requests as defined.